ruby-sdk/ext/eppo_client/src/client.rs

```diff
@@ -237,8 +250,26 @@ impl Client {
     }
 
     pub fn shutdown(&self) {
-        if let Some((thread, _poller)) = self.poller_thread.take() {
-            let _ = thread.graceful_shutdown();
+        if let Some(thread) = self.background_thread.take() {
+            thread.graceful_shutdown();
         }
     }
+
+    pub fn track(&self, event_type: String, payload: Value) -> Result<()> {
+        let Some(event_ingestion) = &self.event_ingestion else {
+            // Event ingestion is disabled, do nothing.
+            return Ok(());
+        };
+
+        let payload: serde_json::Value = serde_magnus::deserialize(payload).map_err(|err| {
```

Comment:
since we're parsing the `payload` as json here, it's easy to move the event validation logic from `EventDelivery` to here now. I'll tackle that next